### PR TITLE
[RLlib] Change `config.fault_tolerance` default behavior (from `recreate_failed_env_runners=False` to `True`).

### PR DESCRIPTION
**Pull request by sven1977**
Change `config.fault_tolerance` default behavior (from `recreate_failed_env_runners=False` to `True`).

The new logic is to always try and restart failed `EnvRunners` by default.

Also renamed `recreate_failed_env_runners` to `restart_failed_env_runners` to match other similar settings and ray's main `max_num_restarts` vocabulary.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/rllib/rllib-training.rst [hunk 1]
```diff
@@ -412,16 +412,16 @@ inference. Make sure to set ``num_gpus: 1`` if you want to use a GPU. If the lea
...
-``recreate_failed_env_runners``: When set to True, your Algorithm will attempt to replace/recreate any failed worker(s) with newly created one(s). This way, your number of workers will never decrease, even if some of them fail from time to time.
-``restart_failed_sub_environments``: When set to True and there is a failure in one of the vectorized sub-environments in one of your workers, the worker will try to recreate only the failed sub-environment and re-integrate the newly created one into your vectorized env stack on that worker.
+``restart_failed_env_runners``: When set to True (default), your Algorithm will attempt to restart any failed EnvRunner and replace it with a newly created one. This way, your number of workers will never decrease, even if some of them fail from time to time.
+``ignore_env_runner_failures``: When set to True, your Algorithm will not crash due to an EnvRunner error, but continue for as long as there is at least one functional worker remaining. This setting is ignored when ``restart_failed_env_runners=True``.
```
Comment (simonsays1980) [inline doc/source/rllib/rllib-training.rst:418]: Why do we want to ignore this setting when `restart_failed_env_runners=True`? Following the semantics a user would understand that if failed env runners are restarted and the algorithm does not ignore failed env runners, the setting has no meaning.

File: doc/source/rllib/rllib-training.rst [hunk 2]
```diff
@@ -412,16 +412,16 @@ inference. Make sure to set ``num_gpus: 1`` if you want to use a GPU. If the lea
...
+``restart_failed_sub_environments``: When set to True and there is a failure in one of the vectorized sub-environments in one of your EnvRunners, RLlib tries to recreate only the failed sub-environment and re-integrate the newly created one into your vectorized env stack on that EnvRunner.
 
-Note that only one of ``ignore_env_runner_failures`` or ``recreate_failed_env_runners`` may be set to True (they are mutually exclusive settings). However,
+Note that only one of ``ignore_env_runner_failures`` or ``restart_failed_env_runners`` should be set to True (they are mutually exclusive settings). However,
```
Comment (simonsays1980) [inline doc/source/rllib/rllib-training.rst:421]: This might need further explanation.

File: rllib/algorithms/algorithm_config.py [hunk 3]
```diff
@@ -3260,11 +3257,11 @@ def fault_tolerance(
                 setting is ignored.
             ignore_env_runner_failures: Whether to ignore any EnvRunner failures
```
Comment (simonsays1980) [inline rllib/algorithms/algorithm_config.py:3258]: Ah, here it becomes clear. Maybe we refer to this section or maybe I overread it above.

**Review by simonsays1980 — APPROVED** [review]
LGTM.

File: doc/source/rllib/rllib-training.rst (hunk 1 above)
Comment (sven1977) [reply doc/source/rllib/rllib-training.rst:418]: If `restart_failed_env_runners=True`, then RLlib doesn't have a choice to a) ignore or b) crash, b/c it has to restart the failed EnvRunner, so this setting (`ignore_env_runner_failures`) becomes irrelevant.

See also: https://docs.ray.io/en/latest/rllib/rllib-training.html#rllib-scaling-guide

**Pull request merged**